crates/sui-bridge/src/sui_syncer.rs

```diff
@@ -118,10 +110,10 @@ where
                     .send((module.clone(), events.data))
                     .await
                     .expect("All Sui event channel receivers are closed");
-                // Unwrap: `query_events_by_module` always returns Some `next_cursor`
-                // If the events list is empty, `next_cursor` will be the same as `start_tx_digest`
-                next_cursor = events.next_cursor.unwrap();
-                tracing::info!(?module, ?next_cursor, "Observed {len} new Sui events");
+                if let Some(next) = events.next_cursor {
```

Comment:
Just curious, when can `events.next_cursor` be none? And if `cursor` is not updated, do the same set of events get returned again?

Reply:
it can be none when the query gets 0 results. Thus it's expected to not update the cursor